src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -75,9 +116,11 @@ def _process_model_before_weight_loading(self, model, **kwargs):
 
         ct_quantization_config = self.compressor.quantization_config
 
-        if self.run_compressed and self.is_quantization_compressed:
+        if self.run_compressed:
+            if not self.is_quantization_compressed:
+                raise ValueError("`run_compressed` is only supported for quantized_compressed models")
             apply_quantization_config(model, ct_quantization_config, run_compressed=True)
-        elif not self.is_quantization_compressed:
+        elif self.is_quantized and not self.is_quantization_compressed:
```

Comment:
Wouldn't any model reaching the HFQuantizer be quantized already?

Reply:
Anything that has a quantization_config, and the quant_method is compressed-tensors would reach this code,
Think about Sparse-Only Compressed cases!